src/FeeRouter/RewardRouter.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+interface IInbox {
+    function createRetryableTicket(
+        address to,
+        uint256 l2CallValue,
+        uint256 maxSubmissionCost,
+        address excessFeeRefundAddress,
+        address callValueRefundAddress,
+        uint256 gasLimit,
+        uint256 maxFeePerGas,
+        bytes calldata data
+    ) external payable returns (uint256);
+}
+
+contract RewardRouter {
+    IInbox immutable inbox;
+    address immutable destination;
+
+    constructor(address _inbox, address _destination) {
+        inbox = IInbox(_inbox);
+        destination = _destination;
+    }
+
+    receive() external payable {}
+
+    function routeFunds(
+        uint256 maxSubmissionCost,
+        uint256 gasLimit,
+        uint256 maxFeePerGas
+    ) external payable {
+        _routeFunds(maxSubmissionCost, gasLimit, maxFeePerGas, msg.sender);
+    }
+
+    function routeFundsCustomRefund(
```

Comment:
Could make this func public and call it from `routeFunds`, then get rid of the `_routeFunds` func